pkg/dynamicaccess/accesslogic.go

```diff
@@ -14,7 +15,7 @@ var hashFunc = sha3.NewLegacyKeccak256
 // Read-only interface for the ACT
 type Decryptor interface {
 	// DecryptRef will return a decrypted reference, for given encrypted reference and grantee
-	DecryptRef(storage kvs.KeyValueStore, encryptedRef swarm.Address, publisher *ecdsa.PublicKey) (swarm.Address, error)
+	DecryptRef(ctx context.Context, storage kvs.KeyValueStore, encryptedRef swarm.Address, publisher *ecdsa.PublicKey) (swarm.Address, error)
```

Comment:
What will this newly added Context contain? The History address?

Reply:
Ctx is either passed down to each component (controller -> actlogic -> kvs) or it is created separately in each component. Not sure yet, which one is the correct implementation.